components/admin_console/custom_plugin_settings/enable_plugin_setting.ts

```diff
@@ -4,7 +4,17 @@ import {Constants} from '../../../utils/constants';
 import {t} from '../../../utils/i18n';
 import SchemaAdminSettings from '../schema_admin_settings';
 
-export default function getEnablePluginSetting(plugin) {
+export type EnabledPluginSetting = {
+    type: string;
+    key: string;
+    label: string;
+    label_default: string;
+    help_text: string;
+    help_text_default: string;
+    isDisabled?: () => boolean;
+}
+
+export default function getEnablePluginSetting(plugin: { id: string }): EnabledPluginSetting {
```

Comment:
Shouldn't the type of `plugin` be `PluginRedux`? Also, is there a less specific type we can use for the return value like `SchemaAdminSettings`?

Reply:
You are right about `plugin`, I changed it. 
For return type, `SchemaAdminSettings` didn't work. I tried to use `PluginSetting` but there was some missing elements. Please see my comments [here](https://github.com/mattermost/mattermost-webapp/pull/10160#discussion_r847123727)